Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Publish-NewServiceFabricApplication.ps1

```diff
@@ -105,7 +105,11 @@
 
         [Parameter(ParameterSetName = "ApplicationParameterFilePath")]
         [Parameter(ParameterSetName = "ApplicationName")]
-        [Switch]$CompressPackage
+        [Switch]$CompressPackage,
+
+        [Parameter(ParameterSetName = "ApplicationParameterFilePath")]
+        [Parameter(ParameterSetName = "ApplicationName")]
+        [ref]$status
```

Comment:
$status is good candidate global variable - then you won't need to pass along.

Reply:
Lets us PassByReference as its more controlled and always safer then to use Global variable

Reply:
Passing an extra optional parameter (with reference) still seems fine to me as there is no extra memory usage

Reply:
In this case there does not seem to be usual problems related to global variables. Before execution a command will set status and that will always be the truth - we don't have multi-threading or parallelization in our flow.
I am in favor of global as code will be cleaner without need to pass $status everywhere.

Reply:
Using global variable now.